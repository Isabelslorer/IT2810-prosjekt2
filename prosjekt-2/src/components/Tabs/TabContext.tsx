import { createContext } from "react";

type tabsContextType = {
    activeTab: number;
    setActiveTab: (value: number) => void;
}

export const tabsContext = createContext<tabsContextType | undefined>(undefined);
