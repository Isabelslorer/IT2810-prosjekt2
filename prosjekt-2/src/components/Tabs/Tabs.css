.buttonContainer{
    display: felx;
    justify-content: space-evenly;
}


.button {
    background-color: #00B0FF;
    width: 150px;
    height: 40px; 
    border: none;
    color: white;
    text-align: center;
    font-size: 14px;
    font-family: Verdana;
    margin: 4px 2px;
    opacity: 0.7;
    transition: 0.3s;
    display: inline-block;
    text-decoration: none;
    cursor: pointer;
}

.button:hover {
    background-color: #059ee6;
    opacity: 1;
}

  @media only screen and (max-width: 600px) {
    .buttonContainer{
      flex-direction: row;
    }
    .button{
        width: 50px;
        height: 20px; 
        font-size: 10px;
    } 
  }