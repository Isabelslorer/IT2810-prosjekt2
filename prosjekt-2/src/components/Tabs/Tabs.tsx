import React, { useContext } from "react";
import {tabsContext} from "./TabContext";
import "./Tabs.css";


function Tabs() {

    const tabContext = useContext(tabsContext);

    return (
        <div className="buttonContainer">
            <button className="button" onClick={() => {sessionStorage.setItem('lastPictureTab', tabContext!.activeTab.toString()); tabContext?.setActiveTab(1)}}>1</button>
            <button className="button" onClick={() => {sessionStorage.setItem('lastPictureTab', tabContext!.activeTab.toString()); tabContext?.setActiveTab(2)}}>2</button>
            <button className="button" onClick={() => {sessionStorage.setItem('lastPictureTab', tabContext!.activeTab.toString()); tabContext?.setActiveTab(3)}}>3</button>
        </div>
    );
 }
  


export default Tabs;