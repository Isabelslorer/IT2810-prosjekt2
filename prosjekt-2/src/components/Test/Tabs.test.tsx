import React from 'react';
import renderer from 'react-test-renderer';

import Tabs from '../../components/Tabs/Tabs';

it("renders correctly when there are no items", () => {
    const tree = renderer.create(<Tabs/>).toJSON();
    expect(tree).toMatchSnapshot();
});