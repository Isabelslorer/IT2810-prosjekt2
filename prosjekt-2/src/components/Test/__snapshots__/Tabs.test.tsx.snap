// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="buttonContainer"
>
  <button
    className="button"
    onClick={[Function]}
  >
    Bilde 1
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Bilde 2
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Bilde 3
  </button>
</div>
`;
